homeassistant/components/input_datetime.py

```diff
@@ -43,8 +43,8 @@
     DOMAIN: vol.Schema({
         cv.slug: vol.All({
             vol.Optional(CONF_NAME): cv.string,
-            vol.Required(CONF_HAS_DATE): cv.boolean,
-            vol.Required(CONF_HAS_TIME): cv.boolean,
+            vol.Optional(CONF_HAS_DATE): cv.boolean,
```

Comment:
Please set the default to `False`. Otherwise, `None` is being passed around instead of `False`, which of course works just as `False` in `if self._has_date`, but gives you strange values in the attributes. Also, yay for type safety.